<launch>
	<arg name="model" />
	<arg name="gui" default="False" />
	<param name="robot_description" textfile="$(arg model)" />
	<param name="use_gui" value="$(arg gui)"/>
	<param name="source_list" value="robot_control_node"/>
	<node name="robot_control_node" pkg="r2_robot" type="robot_control_node.py" />
	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <rosparam param="source_list">[control_joint_states]</rosparam>
	</node>
	<node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />
	<node name="rviz" pkg="rviz" type="rviz" args="-d $(find r2_robot)/urdf.rviz" required="true" />
	<node name="publish_marker" pkg="r2_robot" type="publish_marker.py" />
</launch>
